gm4_live_catch/data/gm4_live_catch/functions/fish/cod/target.mcfunction

```diff
@@ -0,0 +1,8 @@
+# run from gm4_live_catch:fish/FISH_TYPE/catch
+
+
+# summon fish at item
+execute at @a[tag=gm4_lc_fisher] as @e[type=item,limit=1,sort=nearest,nbt={Age:0s}] at @s run function gm4_live_catch:fish/cod/summon
```

Comment:
You're not verifying that the item you're targeting is indeed a fish. In combination with the lack of a distance check this could lead to deletion of items by coincidence and is therefore unsafe.
You should verify that the item you're targeting is a fish and -- if possible -- is a fish that was just caught using a fishing rod. 

This also applies to the other types of fish.

Reply:
Additionally, you're not verifying if this player caught cod or a different fish, each fish should get a separate tag (as mentioned in an above comment) in the case that two players fish up two different fish at the same tick. Technically it wouldn't be visually detectable since the item is destroyed, so both players will get _a_ fish, but still.

Reply:
> You should verify that the item you're targeting is a fish and -- if possible -- is a fish that was just caught using a fishing rod.

It is not possible to detect if the fish was caught using a fishing rod. The code below will verify that it was cod that was caught and most likely to be from the fishing rod (not on the ground, count of 1, age of 0, and furthest instead of nearest as players are not standing right next to their fishing bobber), by a player who caught cod specifically.

```suggestion
execute at @a[tag=gm4_lc_fished_cod] as @e[type=item,limit=1,sort=furthest,nbt={Item:{id:"minecraft:cod",Count:1b},OnGround:0b,Age:0s}] at @s run function gm4_live_catch:fish/cod/summon
```